clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1657,12 +1658,17 @@ class SyclKernelFieldChecker : public SyclKernelFieldHandler {
       if (TAL.size() > 5)
         return checkPropertyListType(TAL.get(5), Loc.getBegin());
     }
+    if (IsSIMD && !Util::isSyclType(Ty, "accessor", true /*Tmp*/))
```

Comment:
I think you can make this check first (after the assert) to avoid unnecessarily hitting above code